File: src/main/java/com/demonwav/mcdev/platform/mcp/at/psi/mixins/AtFuncNameMixin.java
/*
 * Minecraft Dev for IntelliJ
 *
 * https://minecraftdev.org
 *
 * Copyright (c) 2016 Kyle Wood (DemonWav)
 *
 * MIT License
 */
package com.demonwav.mcdev.platform.mcp.at.psi.mixins;

import com.intellij.psi.PsiElement;
import org.jetbrains.annotations.NotNull;

public interface AtFuncNameMixin extends PsiElement {

    @NotNull
    PsiElement getNameElement();

    void setFuncName(@NotNull String funcName);

    @NotNull
    String getFuncNameText();
}

File: src/main/kotlin/com/demonwav/mcdev/platform/mcp/at/psi/mixins/AtFuncNameMixin.kt
/*
 * Minecraft Dev for IntelliJ
 *
 * https://minecraftdev.org
 *
 * Copyright (c) 2019 minecraft-dev
 *
 * MIT License
 */

package com.demonwav.mcdev.platform.mcp.at.psi.mixins

import com.demonwav.mcdev.platform.mcp.at.psi.AtElement
import com.intellij.psi.PsiElement

interface AtFuncNameMixin : AtElement {

    val nameElement: PsiElement
    val funcNameText: String

    fun setFuncName(funcName: String)
}
